### ISSUE TITLE
The filter flag showing incorrect figures

### ISSUE THREAD

**Issue by AnnieMungai** [OpenSRP/opensrp-client-path-zeir#108]
v.0.0.8- preview
The filter button is showing incorrect figures - this client has only 3 clients but the filter shows 7 children (user: Demo3) I also checked with Demo1 and the same issue is happening. 
![image](https://user-images.githubusercontent.com/58034047/112288087-0fa31700-8c9e-11eb-8de9-0e93c1f180ec.png)

**Comment by hamza-vd** [comment]
@AnnieMungai 

During my investigation, I discovered that the filter count was being shown correctly but the no. of children shown on the register was incorrect. The reason was that mother details for those children weren't synced. So those children which didn't have their mother's details in the database were not being returned by the main register query.

The missing mother details might be due to the reason that those records were synced with a different location id (the error we had earlier with the sync) and now they are not being returned.  

Anyhow, I have tested the data with new events for both child and mother and they are getting synced. So the above issue will not happen with the newly generated data.
